event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocator.java

```diff
@@ -71,8 +73,23 @@ private static ScheduledExecutorService defaultExecutor(LifeCycleRegistry lifeCy
         return executor;
     }
 
+    // Compute how many slots in a slab will be used to do a default (new slab) allocation
+    private static int defaultAllocationCount(PeekingIterator<Integer> eventSizes) {
```

Comment:
Like the `allocate` method in `ChannelAllocationState`, this method needs to check each message and ensure it is below `MAX_MESSAGE_SIZE`. Considering the large functionality overlap between the two methods, it may be worth abstracting some of this logic into a common helper method.

Reply:
OK went ahead and did this, almost like the result. The defaultAllocationCount function still has a side effect (it "drains" the iterator that is passed to it) so IMO a better design would be to pass it a list of sizes and let it create its own peeking iterator then it would be 100% side-effect free, it would just be a function you call that returns a result for a given list of message sizes. But maybe I am being too picky.